Tests/SemanticTests/invalid_self_type_in_trait_default.flint

```diff
@@ -0,0 +1,28 @@
+// RUN: %flintc %s --verify
+
+struct Implementation {
+  // Self should only be allowed in traits
```

Comment:
This comment is true but the declaration fails due to struct pass-by-value.

Reply:
Solely because it checks that error before the Self check